components/portal/widgets/controllers.js

```diff
@@ -457,53 +459,55 @@ define(['angular'], function(angular) {
     };
 
     var resolveTemplateStatus = function(now, actionStart, actionEnd) {
-      var result = '';
+      if (!now || !actionStart || !actionEnd) {
+        throw new TypeError('now, actionStart, and actionEnd must all be set');
+      }
+
+      if (moment(actionStart).isAfter(actionEnd)) {
+        throw new RangeError('actionStart must be before actionEnd');
+      }
+
       var statuses = [
         {
           name: 'upcoming',
           check: function() {
-            return now < actionStart && now < actionEnd;
+            return moment(now).isBefore(actionStart);
           },
-        }, {
+        },
+        {
           name: 'ongoing',
           check: function() {
-            return now > actionStart && now < actionEnd;
+            return moment(now).isBetween(actionStart, actionEnd);
           },
-        }, {
+        },
+         {
           name: 'lastDay',
           check: function() {
-            return now > actionStart && now < actionEnd
+            return moment(now).isBetween(actionStart, actionEnd)
               && 1 === resolveDaysLeft(now, actionEnd);
           },
-        }, {
+        },
+        {
           name: 'ended',
           check: function() {
-            return now > actionStart && now > actionEnd;
+            return moment(now).isAfter(actionEnd);
           },
         },
       ];
 
-      if (now && actionStart && actionEnd) {
-        angular.forEach(statuses, function(status) {
-          if (status.check()) {
-            result = status.name;
-          }
-        });
-      }
+      var result = '';
+      // TODO: rewrite as Array.prototype.find when IE support is dropped
+      angular.forEach(statuses, function(status) {
+        if (status.check()) {
+          result = status.name;
+        }
+      });
 
       return result;
     };
 
     var resolveDaysLeft = function(now, actionEnd) {
-      var result = -1;
-      var oneDay = 1000 * 60 * 60 * 24;
-
-      if (now && actionEnd && now < actionEnd) {
-        var difference = actionEnd.getTime() - now.getTime();
-        result = Math.ceil((1.0 * difference) / oneDay);
-      }
-
-      return result;
+      return moment(actionEnd).diff(now, 'days');
```

Comment:
I'd want to check to make sure this behavior matches the existing behavior: when there's less than a day left, it should report 1 day.

Reply:
This doesn't exactly match that behavior, fix incoming

Reply:
Updated in a311bb0